### PR TITLE
FS: Fix user authenticated routes 

### PR DESCRIPTION
**Pull request by kyi193**
Hey guys, this was a hot fix for miscellaneous user authentication issues:

Logout now deletes the token cookie in the browser
Routes that require an authenticated user properly redirects to the sign up page
Sign up and log in pages properly redirects users to the dashboard page if there is an authenticated user

I had a huge issue with merging conflicts with dev with my first PR (fix-routing-and-authentication-on-refresh). That PR has been closed now. The issue was that some of the previous commits (which was not currently fixing this issue), was getting causing all sorts of conflicts with my first PR and I was in rebase purgatory for 6 hours. Instead of trying to merge dev -> the last PR, I decided to pull in the remote dev to my dev first and then manually patch things up so that it works. Hopefully there won't be any merge conflicts this time around but if there are I'm going to need some help. 

I apologize for the last messy commit "Big Hunk", I tried to be as organized as possible but it's 4:30 right now and I need some sleep so we can get waiting room sockets to work :(

### CODE REVIEW

**Comment by kyi193** [comment]
Hey guys, so I fixed up my end points so that it uses the passport middleware. I also deleted cors since we won't be needing it anymore. Let me know what you think @skassam21. I didn't put the middleware in logout since all we're doing on logout is deleting the cookie in the browser (correct me if I'm wrong here though!).

users.js (in routes folder)
Screen Shot 2020-11-21 at 12 27 52 PM
userController.js (all we're doing is returning req.user now since the middleware will handle the jwt verification)
Screen Shot 2020-11-21 at 12 28 32 PM

File: client/src/Components/Navbar.js [hunk 1]
```diff
@@ -45,7 +46,7 @@ const Navbar = (props) => {
     history.push(route);
     if (route === '/signup') {
       dispatch({ type: LOGOUT });
-      fetch('/users/logout', { method: 'POST', credentials: 'same-origin' });
+      axios.post('http://localhost:3000/users/logout')
```
Comment (skassam21) [inline client/src/Components/Navbar.js:49]: We should not be writing localhost:3000 however - it should be /users/logout actually 

I do think we should make this an axios call though, and should be where our other API calls are being made

File: client/src/Components/UserInformation.js [hunk 2]
```diff
@@ -74,11 +74,11 @@ const UserInformation = (props) => {
         `http://localhost:3000/users/update/${id}`,
         userData
       );
-      dispatch({ type: USER_LOADED, payload: result.data });
+      dispatch({ type: USER_LOADED, payload: result.data.user });
       // will change to /background (protected route, routes folder)
       history.push('/dashboard');
     } catch (error) {
-      console.error(error);
+      console.log(error);
```
Comment (skassam21) [inline client/src/Components/UserInformation.js:81]: I think we wanted to keep this as an error log

File: client/src/components/LoginSignupWrapper.js [hunk 3]
```diff
@@ -2,34 +2,26 @@ import React, { useContext, useEffect, useCallback } from 'react';
...
+  const getUserInfo = useCallback(async () => {
     try {
-      const res = await getUser();
-      dispatch({
-        type: USER_LOADED,
-        payload: res.data,
-      });
-    } catch (err) {
-      console.error(err);
+      await axios.get('http://localhost:3000/users/')
```
Comment (skassam21) [inline client/src/components/LoginSignupWrapper.js:15]: again we shouldn't be using localhost:3000

File: client/src/routing/PrivateRoute.js [hunk 4]
```diff
@@ -1,11 +1,38 @@
...
+import { USER_LOADED } from '../context/types'
 const PrivateRoute = ({ component: Component, ...rest }) => {
+
+  const { dispatch, state } = useContext(store);
+  const [isBusy, setBusy] = useState(true);
+
+  const getUserInfo = useCallback(async () => {
+
+    try {
+      const result = await axios.get('http://localhost:3000/users/')
```
Comment (skassam21) [inline client/src/routing/PrivateRoute.js:14]: again same issue

File: server/controllers/userController.js [hunk 5]
```diff
@@ -54,21 +58,8 @@ function editUser(req, res) {
...
-    const user = await User.findById(id);
-
-    res.status(200).json({ user, token });
-
-    // Return user interviews here
-  } catch (err) {
-    res.status(401).json({ error: 'Token is not valid' });
-  }
+function getUser(req, res) {
+  res.status(200).json({ user: req.user });
```
Comment (skassam21) [inline server/controllers/userController.js:62]: 👍

File: server/app.js [hunk 6]
```diff
@@ -17,7 +16,6 @@ const questions = require('./routes/questions');
 const { json, urlencoded } = express;
 
 var app = express();
-app.use(cors({ origin: 'http://localhost:3000', credentials: true }));
```
Comment (skassam21) [inline server/app.js:20]: 👍

File: server/routes/users.js [hunk 7]
```diff
@@ -11,17 +11,12 @@ router.post('/register', userController.register);
 // POST /users/login
 router.post('/login', userController.login);
 
-router.post(
-  '/logout',
-  passport.authenticate('jwt', { session: false }),
-  userController.logout
-);
+// POST /users/logout
+router.post('/logout', userController.logout);
```
Comment (skassam21) [inline server/routes/users.js:15]: this route needs jwt authentication as well

**Comment by kyi193** [comment]
I changed it so that we're using the api functions for axios calls and not use localhost:3000 anymore. I also added the jwt authentication on user logout as well.

**Review by skassam21 — APPROVED** [review]

**Pull request merged**